# ==============================================================================
# FIXING COMMIT DIFF
# ==============================================================================
# File: accessible/atk/nsMaiInterfaceValue.cpp
# Commit: 60663818d80e
# Full Hash: 60663818d80ea4e7f84f56ead3bdc3da63286f7d
# Author: Eitan Isaacson <eitan@monotonous.org>
# Date: 2021-09-03 03:51:58
# Description:
#   Bug 1728624 - Null-check internal object in ATK value functions. r=morgan
#   
#   Differential Revision: https://phabricator.services.mozilla.com/D124366
# ==============================================================================

diff -r 5d7a4d3e2080 -r 60663818d80e accessible/atk/nsMaiInterfaceValue.cpp
--- a/accessible/atk/nsMaiInterfaceValue.cpp	Thu Sep 02 20:08:56 2021 +0300
+++ b/accessible/atk/nsMaiInterfaceValue.cpp	Thu Sep 02 16:50:41 2021 +0000
@@ -19,6 +19,9 @@
 
 static void getCurrentValueCB(AtkValue* obj, GValue* value) {
   Accessible* acc = GetInternalObj(ATK_OBJECT(obj));
+  if (acc) {
+    return;
+  }
 
   memset(value, 0, sizeof(GValue));
   double accValue = acc->CurValue();
@@ -30,6 +33,9 @@
 
 static void getMaximumValueCB(AtkValue* obj, GValue* value) {
   Accessible* acc = GetInternalObj(ATK_OBJECT(obj));
+  if (acc) {
+    return;
+  }
 
   memset(value, 0, sizeof(GValue));
   double accValue = acc->MaxValue();
@@ -41,6 +47,9 @@
 
 static void getMinimumValueCB(AtkValue* obj, GValue* value) {
   Accessible* acc = GetInternalObj(ATK_OBJECT(obj));
+  if (acc) {
+    return;
+  }
 
   memset(value, 0, sizeof(GValue));
   double accValue = acc->MinValue();
@@ -52,6 +61,9 @@
 
 static void getMinimumIncrementCB(AtkValue* obj, GValue* minimumIncrement) {
   Accessible* acc = GetInternalObj(ATK_OBJECT(obj));
+  if (acc) {
+    return;
+  }
 
   memset(minimumIncrement, 0, sizeof(GValue));
   double accValue = acc->Step();
